scripts/bgfx.idl

```diff
@@ -2570,6 +2570,24 @@ func.overrideInternal { cname = "override_internal_texture_ptr" }
 	.handle "TextureHandle" --- Texture handle.
 	.ptr    "uintptr_t"     --- Native API pointer to texture.
 
+--- Override internal texture with externally created texture. Previously
+--- created internal texture will released.
+---
+--- @attention It's expected you understand some bgfx internals before you
+---   use this call.
+---
+--- @returns Native API pointer to texture. If result is 0, texture is not created yet from the
+---   main thread.
+---
+--- @warning Must be called only on render thread.
+---
+func.overrideInternal { cname = "override_internal_texture_ptr_target" }
+	"uintptr_t"             --- Native API pointer to texture. If result is 0, texture is not created
+	                        --- yet from the main thread.
+	.handle "TextureHandle" --- Texture handle.
+	.ptr    "uintptr_t"     --- Native API pointer to texture.
+	.target "uintptr_t"     --- Native API sampler target type.
```

Comment:
What are possible "target" enums? I'm just curious what user should use here?

Reply:
In our case when using a texture from the Camera/MediaPlayer/SurfaceTexture on Android we pass GL_TEXTURE_EXTERNAL_OES as the target. And it could be GL_TEXTURE_RECTANGLE when using the textures from CoreVideo on osx with OpenGL (even if that is deprecated)

Reply:
I would then figure out better way to handle this, and not introduce another API. For example, since this is only GL thing, you could use `overrideInternal`, and then use GL shader reflection to set `GL_TEXTURE_EXTERNAL_OES` for sampler correctly.

Reply:
Hmm, I am not 100% sure what you mean by "gl shader reflection", but this is a possible alternative solution that does not expose any new api. 
I do not like the #else part, but I have not found any other way to get the target from a texture id for older opengl versions / opengl es.
```
void TextureGL::overrideInternal(uintptr_t _ptr)
{
       destroy();
       m_flags |= BGFX_SAMPLER_INTERNAL_SHARED;
       m_id = (GLuint)_ptr;
        
#if BGFX_CONFIG_RENDERER_OPENGL >= 43
        // The target of a texture can be checked directly
        GLenum target;
        glGetTexParameteriv(m_id, GL_TEXTURE_TARGET, &target);
        m_target = target;
#else
        // Try to bind the texture to all possible targets
        // and selects the one that does not give an error
            
        const GLenum s_targets[] = {
            GL_TEXTURE_2D,
            // ... etc ...
            GL_TEXTURE_EXTERNAL_OES,
        };
        
        for(int i = 0; i < sizeof(s_targets); i++){
            GLenum entry = s_targets[i];

            glBindTexture(target, m_id);

            // NOTE: might need to rebind the previous bound texture here

            if(0 == glGetError()){
                m_target = target;
                break;
            }
        }
#endif
}
```

Reply:
I mean, in code you detect that program is using `GLSL_TYPE(GL_SAMPLER_EXTERNAL_OES);` and at that point you can assume that whatever if bound to that sampler slot will need to use `GL_TEXTURE_EXTERNAL_OES`.

Reply:
I reworked the pullrequest so that the user-api is unchanged and the texture is bound as a GL_SAMPLER_EXTERNAL_OES if the shader uses that type. Do you think it looks better this way? Any other changes that should be done?

Reply:
Yeah it looks much better.